WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt

```diff
@@ -353,110 +128,33 @@ class ImprovedMySiteFragment : Fragment(),
-                        val mediaId = data.getLongExtra(MediaPickerConstants.EXTRA_MEDIA_ID, 0)
-                        viewModel.handleSelectedSiteIcon(mediaId)
-                    }
-                    data.hasExtra(MediaPickerConstants.EXTRA_MEDIA_URIS) -> {
-                        val mediaUriStringsArray = data.getStringArrayExtra(
-                                MediaPickerConstants.EXTRA_MEDIA_URIS
-                        ) ?: return
-
-                        val source = PhotoPickerMediaSource.fromString(
-                                data.getStringExtra(MediaPickerConstants.EXTRA_MEDIA_SOURCE)
-                        )
-                        val iconUrl = mediaUriStringsArray.getOrNull(0) ?: return
-                        viewModel.handleTakenSiteIcon(iconUrl, source)
-                    }
-                    else -> {
-                        AppLog.e(
-                                UTILS,
-                                "Can't resolve picked or captured image"
-                        )
-                    }
-                }
-            }
-            RequestCodes.STORIES_PHOTO_PICKER,
-            RequestCodes.PHOTO_PICKER -> if (resultCode == Activity.RESULT_OK) {
-                viewModel.handleStoriesPhotoPickerResult(data)
-            }
-            UCrop.REQUEST_CROP -> {
-                if (resultCode == UCrop.RESULT_ERROR) {
-                    AppLog.e(
-                            MAIN,
-                            "Image cropping failed!",
-                            UCrop.getError(data)
-                    )
-                }
-                viewModel.handleCropResult(UCrop.getOutput(data), resultCode == Activity.RESULT_OK)
-            }
-            RequestCodes.DOMAIN_REGISTRATION -> if (resultCode == Activity.RESULT_OK) {
-                viewModel.handleSuccessfulDomainRegistrationResult(data.getStringExtra(RESULT_REGISTERED_DOMAIN_EMAIL))
-            }
-            RequestCodes.CREATE_SITE -> {
-                viewModel.startQuickStart()
-            }
-            RequestCodes.SITE_PICKER -> {
-                if (data.getIntExtra(WPMainActivity.ARG_CREATE_SITE, 0) == RequestCodes.CREATE_SITE) {
-                    viewModel.startQuickStart()
-                }
-            }
-        }
+        childFragmentManager.findFragmentByTag(CONTENT_FRAGMENT_TAG)?.onActivityResult(requestCode, resultCode, data)
     }
 
-    private fun loadData(items: List<MySiteItem>) {
-        recycler_view.setVisible(true)
+    private fun loadData() {
+        content_container.setVisible(true)
         actionable_empty_view.setVisible(false)
-        (recycler_view.adapter as? MySiteAdapter)?.loadData(items)
+    }
+
+    private fun showFragment(fragment: Fragment) {
+        val fragmentTransaction: FragmentTransaction = childFragmentManager.beginTransaction()
+        childFragmentManager.findFragmentByTag(CONTENT_FRAGMENT_TAG)?.let { previousFragment ->
```

Comment:
❓ I'm wondering if we need to remove the fragmenty manually, doesn't `replace` take care of it?